test/server/cards/01_SOR/units/GideonHaskRuthlessLoyalist.spec.ts

```diff
@@ -0,0 +1,45 @@
+describe('Gideon Hask, Ruthless Loyalist', function () {
+    integration(function (contextRef) {
+        describe('Gideon Hask\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['wampa', 'battlefield-marine'],
+                    },
+                    player2: {
+                        hand: ['vanquish'],
+                        groundArena: ['gideon-hask#ruthless-loyalist', 'specforce-soldier', 'atst'],
+                    }
+                });
+            });
+
+            it('should give an Experience to a friendly unit', function () {
+                const { context } = contextRef;
+
+                // Defeating an enemy unit
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.atst);
+
+                expect(context.player2).toBeAbleToSelectExactly([context.specforceSoldier, context.gideonHask, context.atst]);
+                context.player2.clickCard(context.gideonHask);
+
+                expect(context.gideonHask).toHaveExactUpgradeNames(['experience']);
+            });
+
+            it('should give an Experience to a friendly unit when enemy unit is defeated by event', function () {
```

Comment:
Do you mind consolidating these cases into one? It's bad test practice but we're in a bit of an unfortunate situation because the Jasmine tests are very slow, and we have a limited amount of github actions time in the free tier. So we try to consolidate into as few cases as we reasonably can. We plan to migrate test frameworks eventually to alleviate this issue.

You can look at the tests for Blizzard Assault AT-AT as an example of how we typically set this up.

Reply:
No problem, I moved all test cases into a single unit test